eng/cake/dotnet.cake

```diff
@@ -151,114 +151,6 @@ Task("dotnet-samples")
-            { "CustomBeforeMicrosoftCSharpTargets", MakeAbsolute(File("./src/Templates/TemplateTestExtraTargets.targets")).FullPath },
-            //Try not restore dependecies of 6.0.10
-            { "DisableTransitiveFrameworkReferenceDownloads",  "true" },
-        };
-
-        var templates = new Dictionary<string, Action<DirectoryPath>> {
-            { "maui:maui", null },
-            { "mauiblazor:maui-blazor", null },
-            { "mauilib:mauilib", null },
-            { "mauicorelib:mauilib", dir => {
-                CleanDirectories(dir.Combine("Platforms").FullPath);
-                ReplaceTextInFiles($"{dir}/*.csproj", "UseMaui", "UseMauiCore");
-                ReplaceTextInFiles($"{dir}/*.csproj", "SingleProject", "EnablePreviewMsixTooling");
-            } },
-            { "mauiunpackaged:maui", dir => {
-                ReplaceTextInFiles($"{dir}/*.csproj", "<UseMaui>true</UseMaui>", "<UseMaui>true</UseMaui><WindowsPackageType>None</WindowsPackageType>");
-            } },
-            { "mauiblazorunpackaged:maui-blazor", dir => {
-                ReplaceTextInFiles($"{dir}/*.csproj", "<UseMaui>true</UseMaui>", "<UseMaui>true</UseMaui><WindowsPackageType>None</WindowsPackageType>");
-            } },
-        };
-
-        var alsoPack = new [] {
-            "mauilib"
-        };
-
-        foreach (var template in templates)
-        {
-            foreach (var forceDotNetBuild in new [] { true, false })
-            {
-                // macOS does not support msbuild
-                if (!IsRunningOnWindows() && !forceDotNetBuild)
-                    continue;
-
-                var type = forceDotNetBuild ? "DotNet" : "MSBuild";
-                var projectName = template.Key.Split(":")[0];
-                var templateName = template.Key.Split(":")[1];
-
-                var framework = string.IsNullOrWhiteSpace(TestTFM) ? "" : $"--framework {TestTFM}";
-
-                projectName = $"{tempDir}/{projectName}_{type}";
-                projectName += string.IsNullOrWhiteSpace(TestTFM) ? "" : $"_{TestTFM.Replace('.', '_')}";
-
-                // Create
-                StartProcess(dn, $"new {templateName} -o \"{projectName}\" {framework}");
-
-                // Modify
-                if (template.Value != null)
-                    template.Value(projectName);
-
-                // Enable Tizen
-                ReplaceTextInFiles($"{projectName}/*.csproj",
-                    "<!-- <TargetFrameworks>",
-                    "<TargetFrameworks>");
-                ReplaceTextInFiles($"{projectName}/*.csproj",
-                    "</TargetFrameworks> -->",
-                    "</TargetFrameworks>");
-
-                // Build
-                RunMSBuildWithDotNet(projectName, properties, warningsAsError: true, forceDotNetBuild: forceDotNetBuild, binlogPrefix: "template-");
```

Comment:
In the old cake script, it built all 4 TargetFrameworks.

Does it still _build_ the project for Windows & MacCatalyst? (We don't have to launch apps for these yet)

Reply:
The template build tests will enable Tizen in the project file and then should be building everything. These run on both macOS and Windows hosts to cover all target frameworks. The template run tests will explicitly pass `net7.0-android` or `net7.0-ios` and only build the particular framework that we want to run.

I did have one question about how the template build tests were structured before however -- I wasn't entirely sure if some tests were attempting to build with both `msbuild` and `dotnet build`? The new template build tests only use `dotnet build foo.csproj`.

Reply:
This might be leftover from when you couldn't `dotnet build` a Windows App SDK project.

@mattleibow should we add a build test that runs with `MSBuild.exe`? I vaguely remember that caught an issue once.

Reply:
I've spent some time looking through logs and it can't seem to find any usage of msbuild.exe in the template tests, maybe we should try to add this in a future PR?

Reply:
Yes, if it's not there now, we can add it in another PR.